### PR TITLE
Don't crash with missing class if action is not defined when opening new case

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
If you try and open a case without specifying the action you get an error about missing class because the classname is concatenated based on the activity type name.  But the activity type name is not defined if the action is not specified.  eg. http://localhost:8080/civicrm/case/add?reset=1&atype=13&context=standalone

Before
----------------------------------------
Crash if action not defined. Inconsistent checking for $this->_activityTypeName in some functions (eg. postProcess) but not others.

After
----------------------------------------
Action defaults to ADD. Check if $this->_activityTypeName is defined before trying to use it.

### CODE REVIEW

File: CRM/Case/Form/Case.php [hunk 1]
```diff
@@ -305,8 +315,6 @@ public static function formRule($values, $files, $form) {
    * Process the form submission.
    */
   public function postProcess() {
-    $transaction = new CRM_Core_Transaction();
-
```
Comment (colemanw) [inline CRM/Case/Form/Case.php]: Are you sure this is safe to remove? I'm not exactly sure how transactions work but once you create one it does stuff even if you don't reference the variable, right?

Comment (mattwire) [reply CRM/Case/Form/Case.php]: @colemanw I think you are correct.  I just saw an "unused" variable in phpstorm and removed the line.  I've now "unremoved" it in this PR.

**Comment by colemanw** [comment]
@civicrm-builder retest this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Case/Form/Case.php [hunk 2]
```diff
@@ -87,6 +87,9 @@ class CRM_Case_Form_Case extends CRM_Core_Form {
    * Build the form object.
    */
   public function preProcess() {
+    if (empty($this->_action)) {
+      $this->_action = CRM_Core_Action::ADD;
+    }
```
[the remark this answers is not in the document]
Comment (mattwire) [reply CRM/Case/Form/Case.php]: @jitendrapurohit Yes it is enough.  The other bits were done before I made this change but we don't need them - I've removed them.

**Comment by mattwire** [comment]
@jitendrapurohit Thanks for review, can you test again and confirm if we are ok to merge?

**Comment by jitendrapurohit** [comment]
Sure, tested and confirmed that the PR fixes the main issue here 👍

**Review by jitendrapurohit — APPROVED** [review]

**Pull request merged**